core/psh/pshapp/pshapp.c

```diff
@@ -975,6 +980,9 @@ static int psh_run(int exitable)
 	int err, n, argc;
 	pid_t pgrp;
 
+	while (psh_ttyopen("/dev/tty") < 0)
```

Comment:
You forgot about one feature - allow tty path to be specified as an argument

Also - make default path as DEFINE - it's specified two times in the code